### DISCUSSION TITLE
Element app not allowed as child of element body

### DISCUSSION THREAD

**Discussion by artmedia** [oqtane/oqtane.framework#1924] [General] [answered]
Hello,
after correction in #1901, I was checking the html generated by Oqtane in W3C Validator
https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fwww.artmedia.it%2F

The report shows one html error: "Element app not allowed as child of element body".
The tag < app > in _Host.cshtml seems not correct.
I was wondering if in future versions we can modify it as div: < div id="app" >
I changed it in my local instance and it seems ok.
Thanks

**Comment by sbwalker** [comment, marked as answer]
@artmedia thank you for raising this item... Oqtane is using the approach from the original Blazor Server template:

```
    <app>
        <component type="typeof(Oqtane.App)" render-mode="@Model.RenderMode" param-AntiForgeryToken="@Model.AntiForgeryToken" param-Runtime="@Model.Runtime" param-RenderMode="@Model.RenderMode.ToString()" param-VisitorId="@Model.VisitorId" />
    </app>
```

However I see that in the latest Blazor Server template, it no longer uses an app wrapper tag at all:

```
<component type="typeof(App)" render-mode="ServerPrerendered" />
```

It appears this was removed a while ago... https://github.com/dotnet/aspnetcore/issues/11013
